xml/admin-keystone.xml

```diff
@@ -1821,11 +1838,123 @@ user_enabled_default = 512</screen>
+            <procedure>
+             <step>
+              <para>
+               Add the following lines to
+               <filename>/etc/keystone/keystone.d/999_keystone.conf</filename>:
+              </para>
+              <screen>[ldap]
+debug_level = 56</screen>
+             </step>
+             <step>
+              <para>
+               Restart Keystone and clear the Keystone logs:
+              </para>
+              <screen>$ systemctl stop apache2
+$ sleep 10
+$ ps -ef | grep keystone
+$ > /var/log/keystone/keystone.log
+$ > /var/log/apache2/keystone-admin_error.log
+$ systemctl start apache2></screen>
+             </step>
+             <step>
+              <para>
+               After Keystone restarts, run <command>$ openstack user list
+               --domain <replaceable>DOMAIN_NAME</replaceable></command>.
+              </para>
+             </step>
+             <step>
+              <para>
+               Search for returned referrals:
+              </para>
+              <screen>$ grep referrals
+              /var/log/apache2/keystone-admin_error.log \
+| sort | uniq outstanding referrals 1, parent count 0</screen>
+             </step>
+             <step>
+              <para>
+               Referrals can be ignored by adding <literal>chase_referrals =
+               False</literal> to the
+               <filename>/etc/keystone/keystone.d/999-keystone.conf</filename>
+               file.
+              </para>
+              <screen>[ldap]
+debug_level = 56
+chase_referrals = False</screen>
+             </step>
+             <step>
+              <para>
+               Restart Keystone and clear the Keystone logs as before.
+              </para>
+             </step>
+             <step>
+              <para>
+               Run <command>openstack user list --domain
+               <replaceable>DOMAIN_NAME</replaceable></command> and check the
+               logs. There should be zero outstanding referrals.
+              </para>
+             </step>
+             <step>
+              <para>
+               Remove the debug (<literal>debug_level = 56</literal>) from the
+               <filename>999-keystone.conf</filename> file. Leave in the
+               <literal>chase_referrals = False</literal> configuration option.
+              </para>
+             </step>
+             <step>
+              <para>
+               Copy the
+               <filename>/etc/keystone/keystone.d/999-keystone.conf</filename>
+               file to the other controllers.
+              </para>
+             </step>
+            </procedure>
+           </listitem>
+          </itemizedlist>
+         </section>
       </section>
       <section xml:id="upgrading-keystone" xml:base="identity-upgrading">
         <title>Upgrading Keystone</title>
-        <para>As of the Newton release, keystone supports two different approaches to
+        <para>As of the Newton release, Keystone supports two different approaches to
```

Comment:
So this is something I'd like some clarity on. I don't think we have it defined in the style guide, but upstream we have uppercase for the service name (aka: Identity service) and lowercase for the project names (keystone), see: https://docs.openstack.org/doc-contrib-guide/writing-style/openstack-components.html

Do we have this defined or has this been discussed before @csymons-suse @dmpop? Might be worthwhile to sync with upstream.

Reply:
Just checking in with TR :)

Reply:
Okay just quickly checked in with TR, no major opinions but preference to follow the community guidelines (which, FTR, I recognise is completely different to capitalising the proper noun)

Reply:
I just work here =B^) However, if we're really part of the OpenStack Community then we should be consistent with it. 

Actually I never thought about this. I've only used the entity references...all capitalized. So it would seem that it has been discussed before.

Might be a bit of a project to clean this up...
- changing the entity references is simple
- but also correcting the occurrences that are not entities
- probably something that should be done anyway
- my inclination is to keep pushing towards consistency with upstream, based on comments from SUSE decision makers

Reply:
You're right, that would be a lot. Perhaps we create it as a stretch goal for our Cloud 10 clean up? Thoughts @csymons-suse and @dmpop? We can at least ensure that SOC10 is correct, and from there work backwards whenever we notice something and have a JIRA card open for long-term tracking.

Reply:
Thinking outloud, the entities shouldn't be too hard (as you said) but we should be able to perform a find and replace. Although I say that as if we have infinite time to fix these things between the three of us. Either way, let me know what you think and I'm happy to go ahead and create a JIRA card with the three of us as watchers and add this to the Cloud 10 spec.

Reply:
keystone not changed to initial capital. Many occurrences of lc initial letter.